app/app.go

```diff
@@ -708,6 +735,25 @@ func New(
 
 	app.SetEndBlocker(app.EndBlocker)
 
+	planName := "v0.8.0"
+	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+	})
+
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(fmt.Sprintf("failed to read upgrade info from disk %s", err))
+	}
+
+	if upgradeInfo.Name == planName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		storeUpgrades := storetypes.StoreUpgrades{
+			Added: []string{icacontrollertypes.StoreKey},
+		}
+
+		// configure store loader that checks if version == upgradeHeight and applies store upgrades
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
```

Comment:
should this upgrade handle have the gravity bridge stuff as well?

Reply:
Is it decided that if gravity-bridge is to release in `v0.8.0`?

Reply:
I don't follow the release plan of Cronos, so I'm not sure

Reply:
for ref from the call today: `v0.8.0` (or `v1.0.0`? 😁) is planned for September... if GB can be finalized before that, it's to be included

Reply:
I guess we can complete the upgrade handler later when release plan is more finalized.